src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -0,0 +1,171 @@
+        return commits.get();
+      }
+    }
+
+    List<GHCommit> commits = reverseSort(repositoryFor(project).listCommits().toList());
+    commitsCache.put(project, commits, tomorrow());
+    return commits;
+  }
+
+  /**
+   * Gets the GitHub project repository.
+   * 
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return {@link GHRepository} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public GHRepository repositoryFor(GitHubProject project) throws IOException {
+    if (repositoryCache.containsKey(project)) {
+      Optional<GHRepository> repository = repositoryCache.get(project);
+      if (repository.isPresent()) {
+        return repository.get();
+      }
+    }
+
+    GHRepository repository = github.getRepository(project.path());
+    repositoryCache.put(project, repository);
+    return repository;
+  }
+
+  /**
+   * Gets the first commit of the repository.
+   * 
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return {@link GHCommit} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public GHCommit firstCommitFor(GitHubProject project) throws IOException {
+    List<GHCommit> commits = commitsFor(project);
+    return commits.get(commits.size() - 1);
+  }
+  
+  /**
+   * Sort the commits collection in the reverse order by commit date. That is the first element is
+   * the latest commit and the last element is the first commit.
+   * 
+   * @return reverse sorted List of type {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  private List<GHCommit> reverseSort(List<GHCommit> commits) throws IOException {
+    return commits.stream()
+        .sorted(compareByCommitDate().reversed())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Define the comparator to compare commit date field from {@link GHCommit}.
+   * 
+   * @return {@link Comparator} of generic type {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  private Comparator<GHCommit> compareByCommitDate() throws IOException {
+    return (o1, o2) -> {
+      try {
+        return o1.getCommitDate().compareTo(o2.getCommitDate());
+      } catch (IOException e) {
+        throw new UncheckedIOException("Could not get date from the commit.", e);
+      }
+    };
+  }
+
+  /**
+   * Get a list of commits which comes after a specific date.
+   * 
+   * @param date which is used as boundary.
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return sub-list of type {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public List<GHCommit> filterByAfterDateFor(Date date, GitHubProject project) throws IOException {
+    return commitsFor(project).stream().filter(commit -> {
```

Comment:
I am not sure that streams take into account the fact that the commits are sorted by the commit date. I think it's better to rewrite it with a simple for-loop. That would speed it up a bit mosts cases (the existing data providers check commits for last several months).

Reply:
I agree.